### PR TITLE
feat: add bridge wallet action

### PR DESCRIPTION
**Pull request by BZahory**
### Motivation
Organically exposing MetaMask Mobile users to MetaMask Bridges could contribute to a company-wide initiative to close the gap between our current and targeted revenue projections. Mobile users currently only contribute [~5% of MetaMask Portfolio transactions](https://mixpanel.com/s/3zHPun) despite [mobile users making up over 62% of page views](https://mixpanel.com/s/1sR9lu); there is an opportunity to capture a vast and wildly underrepresented market here.

### Solution
Add a MetaMask Bridges **wallet** action

## **Related issues**

[METABRIDGE-459](https://consensyssoftware.atlassian.net/browse/METABRIDGE-459)

## **Manual testing steps**

1. Load MetaMask Mobile and set-up/create an account.
2. Selected a supported MetaMask Bridges chain (e.g., Ethereum Mainnet)
3. Tap on the wallet actions button on the bottom-middle of the home screen
   - Verify that the "Bridge" row is present with a matching icon and description
4. Tap on the "Bridge" row
   - Verify that MetaMask Bridges opens in the browser on the selected chain.
5. Go back to the home screen
6. Repeat steps 1-4
   - Verify that the existing tab is reopened in the browser.
7. Switch to an unsupported MetaMask Bridges chain
8. Verify that MetaMask Bridges is no longer an available wallet action

### **Before**

Screenshot 2023-11-21 at 5 45 10 PM

### **After**

Screenshot 2023-11-21 at 5 46 50 PM

#### iPhone SE

Screenshot 2023-11-29 at 5 38 04 PM

#### Happy path flows
https://github.com/MetaMask/metamask-mobile/assets/44588480/24a1d24e-5ce7-43ec-a32f-067150d16800

#### Button does not show on unsupported bridge networks

https://github.com/MetaMask/metamask-mobile/assets/44588480/23f3b712-6585-4107-9c2a-017edb36b0f1

#### Opening bridge with MetaMask Portfolio, Uniswap, and a non-web3 tab open

https://github.com/MetaMask/metamask-mobile/assets/44588480/3ff9b3af-fe1d-4fcf-a099-cb990a117dc0

#### Restarting the app with the same tabs open

https://github.com/MetaMask/metamask-mobile/assets/44588480/a69caa4b-08cd-427f-8c7b-940c23da7fd2

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by BZahory** [comment]
I have read the CLA Document and I hereby sign the CLA

File: app/constants/network.js [hunk 1]
```diff
@@ -21,6 +21,8 @@ export const NETWORKS_CHAIN_ID = {
   BSC: '56',
   POLYGON: '137',
   FANTOM: '250',
+  ZKSYNC: '324',
```
[the remark this answers is not in the document]
Comment (BZahory) [reply app/constants/network.js]: Good catch. Resolved

File: app/components/Views/WalletActions/WalletActions.tsx [hunk 2]
```diff
@@ -140,6 +144,18 @@ const WalletActions = () => {
               {...generateTestId(Platform, WALLET_SWAP)}
             />
           )}
+
+        {isBridgeAllowed(chainId) && (
+          <WalletAction
+            actionTitle={strings('asset_overview.bridge')}
+            actionDescription={strings('asset_overview.bridge_description')}
+            iconName={IconName.Bridge}
```
[the remark this answers is not in the document]
Comment (BZahory) [reply app/components/Views/WalletActions/WalletActions.tsx:152]: Yep, it gets drilled down to this line in `Icon.tsx`:

```
const SVG = assetByIconName[name]

File: app/components/UI/Bridge/utils/useGoToBridge.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+const BRIDGE_URL = `${AppConstants.PORTFOLIO_URL}/bridge`;
+
+export default function useGoToBridge(location: string) {
+  const browserTabs = useSelector((state: any) => state.browser.tabs);
+  const { navigate } = useNavigation();
+
+  return (chainId: string, address?: string) => {
+    const existingBridgeTab = browserTabs.find((tab: BrowserTab) =>
+      tab.url.match(new RegExp(`${BRIDGE_URL}/(?![a-z])`)),
```
[the remark this answers is not in the document]
Comment (BZahory) [reply app/components/UI/Bridge/utils/useGoToBridge.ts]: The last two should match, but you're correct that the first one wouldn't match. Updated to just check for `BRIDGE_URL`.

File: app/components/UI/Bridge/utils/useGoToBridge.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+import type { BrowserTab } from '../../Tokens/types';
+import type { BrowserParams } from '../../../../components/Views/Browser/Browser.types';
+
+const BRIDGE_URL = `${AppConstants.PORTFOLIO_URL}/bridge`;
+
+export default function useGoToBridge(location: string) {
+  const browserTabs = useSelector((state: any) => state.browser.tabs);
+  const { navigate } = useNavigation();
+
+  return (chainId: string, address?: string) => {
```
[the remark this answers is not in the document]
Comment (BZahory) [reply app/components/UI/Bridge/utils/useGoToBridge.ts]: Good call. I like this pattern more.

**Review by micaelae — DISMISSED** [review]

**Review by sethkfman — APPROVED** [review]
LGTM

**Pull request merged**